### PR TITLE
Master into v4

### PR DESCRIPTION
**Pull request by haveanicedavid**
## Description

**_PLEASE_** double check any v4 work before approving this. I realize this diff is huge, but a lot of it I had to reconcile manually, and I'm not super familiar with some of the more recent v4 work

Manual backport of master into hambach containing 3 stories to avoid duplicated conflicts:
https://github.com/regen-network/regen-web/pull/1089
https://github.com/regen-network/regen-web/pull/1085
https://github.com/regen-network/regen-web/pull/1084

Note: the diff here was enormous, and I pretty much had to go over each file. Please, double-check any recent v4 features in the deploy preview.

I also apologize, I didn't think through the implications of some of my recent PRs in terms of how they'd effect this branch. Took like 10 hours to get through these conflicts, hope none of that falls on any of your work.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by mhagel** [comment]
@erikalogie it would be a big help if you could do some testing on this to make sure the new v4 features are intact.
https://deploy-preview-1095--regen-registry.netlify.app/

**Review by mhagel — CHANGES_REQUESTED** [review]
Some broken stuff here. I deleted some duplicate files in the MultiStep create batch flow to get that working again, but still some issues:

- Dates should be blank on CreditBasics step
    - see https://deploy-preview-1100--regen-registry.netlify.app/ecocredits/create-batch 
- Validation is not working on this step

Working V4 stuff:
- Got Create Project flow working again.
- Create and Cancel Sell Order 
- Everything else seems ok, but I stopped testing here. We can team up on this if needed.

File: web-registry/src/features/ecocredit/CreateBatchBySteps/CreateBatchMultiStepForm/CreditBasics.tsx [hunk 1]
```diff
@@ -1,39 +1,39 @@
...
 
-import useQueryProjectsByIssuer from '../../../../hooks/useQueryProjectsByIssuer';
 import { MetadataJSONField } from '../../../../components/molecules';
+import useQueryProjectsByIssuer from '../../../../hooks/useQueryProjectsByIssuer';
 import { useWallet } from '../../../../lib/wallet';
-
+import useSaveProjectSelectedOption from '../hooks/useSaveProjectSelectedOption';
 import useUpdateProjectClass from '../hooks/useUpdateProjectClass';
 import useUpdateProjectOptions from '../hooks/useUpdateProjectOptions';
-import useSaveProjectSelectedOption from '../hooks/useSaveProjectSelectedOption';
```
Comment (mhagel) [inline web-registry/src/features/ecocredit/CreateBatchBySteps/CreateBatchMultiStepForm/CreditBasics.tsx:37]: I can't tell what happened here, but this page is not working right. It was working in this [change](https://github.com/regen-network/regen-web/pull/1065/) , but validation and initial form state are not right.

**Comment by erikalogie** [comment]
**Errors I've found:**
- Create a sell order popup no longer generates a transaction to sign
- Could not complete "put in basket" due to error: https://monosnap.com/file/Lqqr6z7ZuoIK0SxoA7DWuEt8Gs4jaa
- Send confirmation popup is missing title: https://monosnap.com/file/TqekJCFpJng0ITmm21KQzZqnherIMI
- I can't input a State/Region in the Retire or Take or Send popups
- Retire confirmation popup is missing title: https://monosnap.com/file/IqB18mnbbks2SLaojoo9rRXmSFuJVf
- Take from basket confirmation popup is missing title
- Getting NaN for total credits issued on credit batch details page: https://monosnap.com/file/5kgVzRWMeXMSIyJWnkythssaMhhoC5
- Why does the NCT page display the porfolio? We will be getting rid of this navigation item for launch, so not worth fixing, but it is weird nonetheless.
- Weird styling in the create project flow: https://monosnap.com/file/5a4Z3Snu8dPoh4LguaFT6DxsLU6DF2
- https://monosnap.com/file/Is4UqaqEusKIb8A2wFEMkEvf8Ooc8o
- https://monosnap.com/file/b9xTDguaotovWVURMtPdLIMnujbAWp
- https://monosnap.com/file/gCYc9KdnMr5XMZTEBADyN6K8TiEV97
- Create credit batch does not generate a transaction to sign 
- Background color is wrong for all the create flows, should be FAFAFA
- Can't generate a transaction to sign with the create credit batch flow

**Comment by erikalogie** [comment]
@haveanicedavid I'm not sure if this is where I should be commenting on https://deploy-preview-1101--regen-registry.netlify.app/ but here it goes.

It looks like to me that none of the above issues have been addressed, and I also found this one:
- Can't cancel a sell order - https://monosnap.com/file/WfbyZyVQQQ4KJG0xhSwiIhTod4DQNb

> **Errors I've found:**
> 
> * Create a sell order popup no longer generates a transaction to sign
> * Could not complete "put in basket" due to error: https://monosnap.com/file/Lqqr6z7ZuoIK0SxoA7DWuEt8Gs4jaa
> * Send confirmation popup is missing title: https://monosnap.com/file/TqekJCFpJng0ITmm21KQzZqnherIMI
> * I can't input a State/Region in the Retire or Take or Send popups
> * Retire confirmation popup is missing title: https://monosnap.com/file/IqB18mnbbks2SLaojoo9rRXmSFuJVf
> * Take from basket confirmation popup is missing title
> * Getting NaN for total credits issued on credit batch details page: https://monosnap.com/file/5kgVzRWMeXMSIyJWnkythssaMhhoC5
> * Why does the NCT page display the porfolio? We will be getting rid of this navigation item for launch, so not worth fixing, but it is weird nonetheless.
> * Weird styling in the create project flow: https://monosnap.com/file/5a4Z3Snu8dPoh4LguaFT6DxsLU6DF2
> * https://monosnap.com/file/Is4UqaqEusKIb8A2wFEMkEvf8Ooc8o
> * https://monosnap.com/file/b9xTDguaotovWVURMtPdLIMnujbAWp
> * https://monosnap.com/file/gCYc9KdnMr5XMZTEBADyN6K8TiEV97
> * Create credit batch does not generate a transaction to sign
> * Background color is wrong for all the create flows, should be FAFAFA
> * Can't generate a transaction to sign with the create credit batch flow

**Pull request closed without merging**